### ISSUE TITLE
Threema can't add custom field to contact

### ISSUE THREAD

**Issue by 1337sup3rh4x0r** [bitfireAT/davdroid#476]
Hi there,
I've been in contact with the support staff of the secure messaging app Threema. Upon installation this app reads my address book, creates hashes from all telephone numbers and email addresses and compares these hashes to those stored on the Threema servers. If matches are found, the app then wants to write the Threema-ID into a new field. 

For my DAVdroid adress book this is not working and after elaborating this with the Threema support, they think that it must be due to Threema not being able to create this new custom field. 

Might this be a DAVdroid issue that could be fixed here?

**Comment by devvv4ever** [comment]
What kind of field would Threema write. What is the exact name of the field? Is it a CardDAV compatible field?

**Comment by 1337sup3rh4x0r** [comment]
I already asked the Threema support for this info, response pending. 

So if it is a custom field, incompatible to CardDAV format, could it still be written and only not synced (I could live with that) or could it not be written at all? 
The latter would basically make the combination of Threema and CardDAV impossible...

**Comment by rfc2822** [comment]
> If matches are found, the app then wants to write the Threema-ID into a new field.

I guess they're writing a (custom data row with a custom MIME type)(https://developer.android.com/reference/android/provider/ContactsContract.Data.html). Is this correct?

> For my DAVdroid adress book this is not working and after elaborating this with the Threema support, they think that it must be due to Threema not being able to create this new custom field.

Can you provide the contents of your exchange with Threema support?

I can imagine that:

* Threema writes a custom row to the raw contact (which is assigned to a DAVdroid account), thus marking the raw contact as updated ("dirty"). Now, Threema should work correctly with this contact.
* As soon as DAVdroid synchronizes, it uploads the dirty contact to the server (without Threema-specific data). The server may then send an updated contact back, and DAVdroid replaces the contact (including the Threema-specific rows) by the contact from the server (without Threema-specific rows).

Can you please provide exact details on how to reproduce this issue, including [verbose logs](https://github.com/bitfireAT/davdroid/wiki/How-to-view-the-logs)?

Does this problem occur if you disable synchronizing (device setting) before setting up a contact with Threema?

**Comment by rfc2822** [comment]
Unfortunately, I can't find which field that could be and everything seems to work here when trying DAVdroid + Threema.

Can you please provide detailed instructions on how to produce this issue and all your communication with Threema support regarding this problem?

**Comment by rfc2822** [comment]
I have now using DAVdroid + Threema myself and can't see any problems. Closing this until further information arrives.

**Comment by 1337sup3rh4x0r** [comment]
Hi rfc2822,
I am really sorry for not responding for over a month now. Your replies somehow slipped my awareness and I had already arranged myself with not beeing able to sync the contacts from threema. However today this bugged me again and I tried a different app (CardDAV Sync from playstore) for syncing my owncloud contacts. And voilá, syncing worked with that one. So I actually came back here to report this as I figured it might help to figure out which end is faulty here. But now I see you seem to already have looked deeply in to my problem, even installed Threema yourself. Thank you for your support! And again, sorry I didn't reply here...

Anyway, my mails with Threema support are all in German, so I'll try to sum up what I've learned:
Threema creates a new account with its own address book in android upon installation. It stores the threema ID of each contact that was previously matched in a field "sync1" within this Threema address book.

But since you report that it just works I will now try and remove all addressbooks as well as Threema and start from scratch. Maybe this helps. This time I'll report back! ;)
